src/app/modules/item/pages/item-log-view/item-log-view.component.ts

```diff
@@ -51,16 +40,51 @@ export class ItemLogViewComponent implements OnChanges {
   ) {}
 
   ngOnChanges(): void {
-    if (this.itemData) this.id$.next(this.itemData.item.id);
+    if (!this.itemData) {
+      return;
+    }
+
+    this.id$.next(this.itemData.item.id);
+    this.logColumns = this.getLogColumns(this.itemData.item.type);
   }
 
-  getData$(id: string): Observable<Data> {
+  private getData$(id: string): Observable<Data> {
     return this.activityLogService.getActivityLog(id).pipe(
       map((data: ActivityLog[]) => ({
-        columns: logColumns,
+        columns: this.logColumns || [],
```

Comment:
If it is just to cover your initial case when logColumn may be undefined even it cannot happen in practice `getLogColumns` would be called first, just don't make `logColumns` optional and set its default value to `[]`

Reply:
Fixed